openshift/machine-config-operator
# OCPBUGS-6945: Fixes node OS detection

**- What I did**

The OS release node detection was not correctly detecting RHCOS 9 OSes due to the VERSION_ID field being populated with the OCP version instead of the OS version.

**- How to verify it**

1. Run the unit test suite.
2. Deploy an OCP cluster running either RHCOS 8 or RHCOS 9 (for RHCOS 9, provision a cluster using ClusterBot pointed at https://github.com/openshift/machine-config-operator/pull/3485).
3. Run the e2e test suite. The e2e test will output additional log entries indicating the node OS and, in the case of RHCOS 8 or 9, the version it is running, which should match the `RHEL_VERSION` field in the `/etc/os-release` / `/usr/lib/os-release` files on the node.

These logs will resemble:
```console
go test -v -count=1 -run="TestOSDetection" .
=== RUN   TestOSDetection
=== RUN   TestOSDetection/#00
=== PAUSE TestOSDetection/#00
=== RUN   TestOSDetection/#01
=== PAUSE TestOSDetection/#01
=== RUN   TestOSDetection/#02
=== PAUSE TestOSDetection/#02
=== RUN   TestOSDetection/#03
=== PAUSE TestOSDetection/#03
=== RUN   TestOSDetection/#04
=== PAUSE TestOSDetection/#04
=== RUN   TestOSDetection/#05
=== PAUSE TestOSDetection/#05
=== CONT  TestOSDetection/#00
=== CONT  TestOSDetection/#03
=== CONT  TestOSDetection/#02
=== CONT  TestOSDetection/#04
=== CONT  TestOSDetection/#01
=== CONT  TestOSDetection/#05
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-173-134.ec2.internal with version 8.6
=== CONT  TestOSDetection/#04
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-163-214.ec2.internal with version 8.6
=== CONT  TestOSDetection/#01
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-138-110.ec2.internal with version 8.6
=== CONT  TestOSDetection/#00
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-134-197.ec2.internal with version 8.6
=== CONT  TestOSDetection/#02
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-147-135.ec2.internal with version 8.6
=== CONT  TestOSDetection/#03
    mcd_test.go:55: Identified OS "Red Hat Enterprise Linux CoreOS" on node ip-10-0-152-25.ec2.internal with version 8.6
--- PASS: TestOSDetection (0.08s)
    --- PASS: TestOSDetection/#05 (0.67s)
    --- PASS: TestOSDetection/#04 (0.67s)
    --- PASS: TestOSDetection/#01 (0.70s)
    --- PASS: TestOSDetection/#00 (0.73s)
    --- PASS: TestOSDetection/#02 (0.74s)
    --- PASS: TestOSDetection/#03 (0.83s)
PASS
ok      github.com/openshift/machine-config-operator/test/e2e   1.566s
```


**- Description for the changelog**
Fixes node OS detection
